DRIVERS-2927 Make `trimFactor` and `sparsity` optional (#1613)
mongodb/specifications: source/client-side-encryption/tests/README.md

```

Assert that an error was raised.

### 22. Range Explicit Encryption applies defaults

> This test requires libmongocrypt with changes of [MONGOCRYPT-698](https://jira.mongodb.org/browse/MONGOCRYPT-698).

Reviewer: ```suggestion
This test requires libmongocrypt with changes in [14ccd9ce](https://github.com/mongodb/libmongocrypt/commit/14ccd9ce8a030158aec07f63e8139d34b95d88e6) ([MONGOCRYPT-698](https://jira.mongodb.org/browse/MONGOCRYPT-698)).
```
Recommend linking to the required commit in addition to the JIRA ticket.

#### Test Setup

Create a MongoClient named `keyVaultClient`.

Create a ClientEncryption object named `clientEncryption` with these options:

```typescript
class ClientEncryptionOpts {
   keyVaultClient: keyVaultClient,
   keyVaultNamespace: "keyvault.datakeys",
   kmsProviders: { "local": { "key": "<base64 decoding of LOCAL_MASTERKEY>" } },
}
```

Create a key with `clientEncryption.createDataKey`. Store the returned key ID in a variable named `keyId`.

> Call `clientEncryption.encrypt` to encrypt the int32 value 123 with these options:

Reviewer 2: ```suggestion
Call `clientEncryption.encrypt` to encrypt the int32 value `123` with these options:
```

Formatting.

```typescript
class EncryptOpts {
   keyId : keyId,
   algorithm: "Range",
   contentionFactor: 0,
   rangeOpts: RangeOpts {
      min: 0,
      max: 1000
   }
}
```

Store the result in a variable named `payload_defaults`.

#### Case 1: Uses libmongocrypt defaults

> Call `clientEncryption.encrypt` to encrypt the int32 value 123 with these options:

Reviewer: ```suggestion
Call `clientEncryption.encrypt` to encrypt the int32 value `123` with these options:
```
Formatting.

```typescript
class EncryptOpts {
   keyId : keyId,
   algorithm: "Range",
   contentionFactor: 0,
   rangeOpts: RangeOpts {
      min: 0,
      max: 1000,
      sparsity: 2,
      trimFactor: 6
   }
}
```

Assert the returned payload size equals the size of `payload_defaults`.

> > [!NOTE]
> > The payload contents are not compared. The payloads include random data. The `trimFactor` and `sparsity` directly
> > affect the payload size.

Reviewer: ```suggestion
> [!NOTE]
> Do not compare the payload contents. The payloads include random data. The `trimFactor` and `sparsity` directly
> affect the payload size.
```
Imperative language.

Suggestion also applies to note below.

#### Case 2: Accepts `trimFactor` 0

> Call `clientEncryption.encrypt` to encrypt the int32 value 123 with these options:

Reviewer: ```suggestion
Call `clientEncryption.encrypt` to encrypt the int32 value `123` with these options:
```
Formatting.
